duckduckgo/iOS
# Allow other apps to share with our app

Task/Issue URL: https://app.asana.com/0/72649045549333/683523325946469
Tech Design URL:
CC:

**Description**:
• Allows other apps to share text and urls with our app
• Also fixes a bug that garbled the scheme of shared urls turning them into a query (e.g urls pasted via force touch)

**Steps to test this PR**:
**TEST 1**
1. Kill the ddg app
1. Open safari and navigate to a url e.g example.com
1. Share the url with our app
1. Ensure the url opens in a new tab

**TEST 2**
Repeat previous test, this time skipping step 1 (don't kill the app first)

**TEST 3**
1. Kill the ddg app
1. Open safari and navigate to a url e.g example.com
1. Highlight some text e.g "This domain is established" from example.com and share it with our app
1. Ensure the shared text opens as a query in a new tab

**TEST 4**
Repeat previous test, this time skipping step 1 (don't kill the app first)

**TEST 5**
1. From safari, copy a full url to the clipboard
1. Now force touch on our app icon and select "Paste from clipboard"
1. Ensure the url opens (rather than a query on our serp)

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
